pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -88,7 +84,8 @@ func ApplyUnstructuredResourceImproved(
 	}
 	existing, err := client.Resource(resourceGVR).Namespace(namespace).Get(ctx, name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		want, err := client.Resource(resourceGVR).Namespace(namespace).Create(ctx, required, metav1.CreateOptions{})
+		var want *unstructured.Unstructured
```

Comment:
nit: since we are returning from this branch anyway I would remove this variable and just simply `want, err := client`